lib/inferno/dsl/fhir_evaluation/evaluator.rb

```diff
@@ -25,7 +27,7 @@ def evaluate(data, config = Config.new)
           end
 
           Rule.descendants.each do |rule|
-            rule.new.check(context) if active_rules.include?(rule.name.demodulize)
+            context.add_result rule.new.check(context) if active_rules.include?(rule.name.demodulize)
```

Comment:
I'd like to keep each rule adding results to `context.results` itself, rather than just returning a result to be added here, because we might have rules that add multiple messages. Think along the lines of the validator which might return errors and warnings and informational messages all at the same time.

In practice the only rule with multiple messages right now is `AllSearchParametersHaveExamples` which can add an extra message if processing the SearchParam fails, but if we add more error handling I think there could be more of that kind of thing

Reply:
Sounds reasonable to me. Put `context.add_result` back to one. https://github.com/inferno-framework/inferno-core/pull/583/commits/c9f171c2b63e1ea56208ccee9839e3ecda3d46c9